istioctl/cmd/kubeinject.go

```diff
@@ -535,7 +535,7 @@ It's best to do kube-inject when the resource is initially created.
 			retval := inject.IntoResourceFile(injector, sidecarTemplate, valuesConfig, rev, meshConfig,
 				reader, writer, func(warning string) {
 					warnings = append(warnings, warning)
-				})
+				}, nil)
```

Comment:
If we only allow it for tests then, IMO we should make `IntoResourceFile` unchanged and make a new internal private `intoResourceFileWithConfig` that `IntoResourceFile` calls with `nil`. Otherwise we leak test detaIls into the API